tilings/strategy_pack.py

```diff
@@ -38,6 +41,47 @@ def replace_list(strats):
             iterative=self.iterative,
         )
 
+    def setup_subclass_verification(self, start_tiling: "Tiling") -> "TileScopePack":
+        """
+        If the subclass verification strategy already has a list of perms to check,
+        we leave it alone. Otherwise we:
+         - compute the cell basis for each cell
+         - consider the permutations avoiding some cell basis
+         - add to the list of perms to check any perm that avoids some cell basis and
+           has length strictly smaller than the maximum length cell basis element.
+        """
+
+        def alter_list(strats) -> None:
+            """
+            Find subclass verification and alter its perms_to_check variable.
+            """
+            for strategy in strats:
+                if isinstance(strategy, strat.SubclassVerificationStrategy):
+                    if strategy.perms_to_check is None:
+                        strategy.perms_to_check = set()
+                        cell_bases = set(
+                            tuple(obs) for obs, _ in start_tiling.cell_basis().values()
+                        )
+                        max_length = (
+                            max(max(len(b) for b in basis) for basis in cell_bases) - 1
+                        )
+                        perm_gen = Perm.up_to_length(max_length)
+                        for perm in perm_gen:
+                            if len(perm) == 0:
+                                continue
+                            if any(perm.avoids(*basis) for basis in cell_bases):
+                                strategy.perms_to_check.add(perm)
+                    logger.info(
+                        "SubclassVerification set up to check the subclasses: Av(%s)",
+                        "), Av(".join(map(str, strategy.perms_to_check)),
+                    )
```

Comment:
If you send a 1x1, e.g. Av(1234) this prints 
Av(0), Av(01), Av(10), Av(012), Av(021), Av(102), Av(120), Av(201), Av(210)
It would be cleaner to say: proper subclasses of Av(0123).